source/dpp/translation/aggregate.d

```diff
@@ -530,18 +530,22 @@ private string[] maybeC11AnonymousRecords(in from!"clang".Cursor cursor,
     const dtype = translate(member.type, context);
     lines ~= "    " ~ dtype ~ " " ~  varName ~ ";";
 
-    foreach(subMember; member.children) {
-        if(subMember.kind == Cursor.Kind.FieldDecl)
-            lines ~= innerFieldAccessors(varName, subMember);
-        else if(subMember.type.canonical.kind == Type.Kind.Record &&
-                hasAnonymousSpelling(subMember.type.canonical) &&
-                !member.children.any!(a => isFieldOfRightType(subMember, a))) {
-            foreach(subSubMember; subMember) {
-                lines ~= "        " ~ innerFieldAccessors(varName, subSubMember);
+    static string[] subMemberAccessors(in Cursor member, in string varName) {
+        string[] res;
+        foreach(subMember; member.children) {
+            if(subMember.kind == Cursor.Kind.FieldDecl)
+                res ~= innerFieldAccessors(varName, subMember);
+            else if(subMember.type.canonical.kind == Type.Kind.Record &&
+                    hasAnonymousSpelling(subMember.type.canonical) &&
+                    !member.children.any!(a => isFieldOfRightType(subMember, a))) {
+                res ~= subMemberAccessors(subMember, varName);
```

Comment:
(This is where the change is, compared to the previous version)
Recursive call on this member if it is another anonymous struct/union, so that we generate accessors no matter the depth of the nesting.